pkgs/applications/audio/pd-plugins/timbreid/default.nix

```diff
@@ -1,30 +1,45 @@
-{ stdenv, fetchurl, unzip, puredata }:
+{ stdenv, fetchurl, unzip, puredata, fftwSinglePrec }:
 
 stdenv.mkDerivation rec {
-  version = "0.6.0";
+  version = "0.7.0";
   name = "timbreid-${version}";
 
   src = fetchurl {
     url = "http://williambrent.conflations.com/pd/timbreID-${version}-src.zip";
-    sha256 = "02rnkb0vpjxrr60c3hryv7zhyjpci2mi9dk27kjxpj5zp26gjk0p";
+    sha256 = "14k2xk5zrzrw1zprdbwx45hrlc7ck8vq4drpd3l455i5r8yk4y6b";
   };
 
-  buildInputs = [ unzip puredata ];
+  buildInputs = [ unzip puredata fftwSinglePrec ];
+
+  makeFlags = [
+    "CFLAGS= -I${puredata}/include/pd -Wall -W -g “-fPIC” --enable-float"
+    "LIBDIR=$(out)/lib"
+  ];
 
   unpackPhase = ''
+    mkdir source
+    cd source
     unzip $src
-    mv timbreID-0.6.0-src/tID/* .
-    rm -rf timbreID-0.6.0-src/tID/
-    rm -rf timbreID-0.6.0-src/INSTALL.txt
   '';
 
+  buildPhase = ''
+    make tIDLib.o all
+ '';
```

Comment:
Are those arguments needed? `tIDLib.o` looks useless. Specify needed arguments in `makeFlags`. By replacing the `buildPhase` you effectively ignore the value of `makeFlags`.

Reply:
without this, I get:
```
building
build flags: SHELL=/nix/store/ski5znw2b8bj5qnsncrcq6128hkrwgvq-bash-4.4-p12/bin/bash LIBDIR=\$\(out\)/lib
gcc -DPD -I/Applications/Pd-0.47-1.app/Contents/Resources/src -I/usr/local/include -Wall -W -g -fPIC -O6 -funroll-loops -fomit-frame-pointer -o "bark.o" -c "bark.c"
gcc  -Wl,--export-dynamic -shared -fPIC -o "bark.pd_linux" "bark.o" tIDLib.o -lfftw3f -lc
gcc: error: tIDLib.o: No such file or directory
make: *** [Makefile:57: bark.pd_linux] Error 1
rm bark.o
builder for ‘/nix/store/22cc8jl0ipn5y4sq3z89p8rbnz7mb6hg-timbreid-0.7.0.drv’ failed with exit code 2
```

Also, the install instructions tell you to do this.